lua/py-requirements/api.lua

```diff
@@ -63,4 +65,53 @@ function M.get_versions(name)
     return versions
 end
 
+---@class ModuleDescription
+---@field content string[]
+---@field content_type? string
+
+---@return ModuleDescription
+function M.parse_description(result)
+    if result == nil or not vim.tbl_contains({ 200, 301 }, result.status) then
+        return { content = {} }
+    end
+    local json = vim.json.decode(result.body)
+    if json == nil or json.info.description == nil then
+        return { content = {} }
+    end
+    return {
+        content = vim.split(json.info.description, '\n'),
+        content_type = json.info.description_content_type,
+    }
+end
+
+---@param module PythonModule
+---@return ModuleDescription
+function M.get_description(module)
```

Comment:
Can we take a name string and an optional version string as parameters rather than the full `PythonModule`.

I'm going to write some unit tests for this module at some point and it'll be easier if I don't need to create the entire structure of a module.

Reply:
One thing to be aware of with the way this is currently implemented is changes to the version won't update the description until vim is restarted based on the way the result is cached.

Not a problem, just something that might come up.

Reply:
Yeah, I had that thought myself as well, whether I wanted to add the version number to the cache key as well. In the end I figured it's an edge case that may just not be worth the hassle until it becomes an issue.